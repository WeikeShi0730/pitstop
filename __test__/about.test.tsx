import { render, RenderResult } from "@testing-library/react";
import About from "../components/about.component";

let documentBody: RenderResult;
describe("<About />", () => {
  beforeEach(() => {
    documentBody = render(<About />);
  });
  it("shows content in <About />", () => {
    const { baseElement } = documentBody;
    expect(baseElement).toMatchSnapshot();
  });
});
