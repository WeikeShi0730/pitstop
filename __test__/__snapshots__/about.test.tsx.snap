// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<About /> shows content in <About /> 1`] = `
<body>
  <div>
    <div
      class="min-h-content flex justify-center items-center text-sm md:text-base"
    >
      <div
        class="m-auto"
      >
        <div
          class="max-w-xs sm:max-w-xl md:max-w-2xl lg:max-w-3xl flex justify-center items-center m-auto my-24 h-full text-slate-700 bg-slate-100 bg-opacity-30 rounded-lg shadow-md hover:shadow-slate-500 transition-all ease-in-out duration-200 flex-col md:flex-row"
        >
          <div
            class="md:-rotate-90"
          >
            <div
              class="relative w-48 h-48 md:w-72 md:h-48 my-5 md:-my-28 md:-translate-y-10"
            >
              <span
                style="box-sizing: border-box; display: block; overflow: hidden; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px; position: absolute; top: 0px; left: 0px; bottom: 0px; right: 0px;"
              >
                <img
                  alt="Logo"
                  class="object-contain rounded-lg"
                  data-nimg="fill"
                  decoding="async"
                  src="/pitstop-logos/pitstop.WebP"
                  style="position: absolute; top: 0px; left: 0px; bottom: 0px; right: 0px; box-sizing: border-box; padding: 0px; margin: auto; display: block; width: 0px; height: 0px; min-width: 100%; max-width: 100%; min-height: 100%; max-height: 100%;"
                />
              </span>
            </div>
          </div>
          <div
            class="md:flex-col leading-8 py-5 px-10 md:-ml-32 text-justify"
          >
            <p>
              <a
                class="underline-primary font-normal text-base md:text-xl"
              >
                Pitstop™
              </a>
               
              is an E-commerce website that sells various F1™ related stickers, including some meme stickers. You can find products of different teams, drivers, and team principles. Our products give you a chance to showcase your love towards F1™. Enjoy!
            </p>
            <p
              class="flex justify-end"
            >
              <a
                class="underline-primary font-normal text-base md:text-xl"
                href="/"
              >
                Go to shop →
              </a>
            </p>
          </div>
        </div>
        <div
          class="max-w-xs sm:max-w-xl md:max-w-2xl lg:max-w-3xl flex justify-center items-center m-auto my-24 h-full text-slate-700 bg-slate-100 bg-opacity-30 rounded-lg shadow-md hover:shadow-slate-500 transition-all ease-in-out duration-200 flex-col md:flex-row-reverse"
        >
          <div>
            <div
              class="relative w-48 h-48 md:w-56 md:h-82 my-5 md:-my-28 rounded-lg shadow-md hover:shadow-slate-500 transition-all ease-in-out duration-200"
            >
              <span
                style="box-sizing: border-box; display: block; overflow: hidden; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px; position: absolute; top: 0px; left: 0px; bottom: 0px; right: 0px;"
              >
                <img
                  alt="My photo"
                  class="object-cover rounded-lg"
                  data-nimg="fill"
                  decoding="async"
                  src="/about/me.WebP"
                  style="position: absolute; top: 0px; left: 0px; bottom: 0px; right: 0px; box-sizing: border-box; padding: 0px; margin: auto; display: block; width: 0px; height: 0px; min-width: 100%; max-width: 100%; min-height: 100%; max-height: 100%;"
                />
              </span>
            </div>
          </div>
          <p
            class="leading-8 py-5 px-10 text-justify"
          >
            My name is
             
            <a
              class="underline-primary font-normal text-base md:text-xl"
            >
              Weike
            </a>
            , and I'm a HUGE Formula 1 fan. Also, I'm a passionate
             
            <a
              class="underline-primary font-normal text-base md:text-xl"
            >
              web developer
            </a>
            . This website is created using some of the most trending technologies that help with web development and UI/UX. Such as
             
            <a
              class="underline-primary font-normal text-base md:text-xl"
            >
              Next.js
            </a>
            ,
             
            <a
              class="underline-primary font-normal text-base md:text-xl"
            >
              React
            </a>
            ,
             
            <a
              class="underline-primary font-normal text-base md:text-xl"
            >
              TailwindCSS
            </a>
            ,
             
            <a
              class="underline-primary font-normal text-base md:text-xl"
            >
              Typescript
            </a>
            ,
             
            <a
              class="underline-primary font-normal text-base md:text-xl"
            >
              Firebase
            </a>
            , and
             
            <a
              class="underline-primary font-normal text-base md:text-xl"
            >
              Stripe
            </a>
            . They ensure to bring a smooth and quick web browsing experience.
          </p>
        </div>
        <div
          class="max-w-xs sm:max-w-xl md:max-w-2xl lg:max-w-3xl flex justify-center items-center m-auto my-24 h-full text-slate-700 bg-slate-100 bg-opacity-30 rounded-lg shadow-md hover:shadow-slate-500 transition-all ease-in-out duration-200"
        >
          <p
            class="leading-8 py-5 px-10 text-justify"
          >
            I love to learn new skills and am passionate about creating web applications to solve real problems and find solutions to human laziness using the latest tech. You can find me on
             
            <a
              class="underline-primary font-normal text-base md:text-xl"
              href="https://github.com/WeikeShi0730"
              rel="noopener noreferrer"
              target="_blank"
            >
              Github
            </a>
            ,
             
            <a
              class="underline-primary font-normal text-base md:text-xl"
              href="https://www.linkedin.com/in/weike-shi/"
              rel="noopener noreferrer"
              target="_blank"
            >
              LinkedIn
            </a>
            ,
             
            <a
              class="underline-primary font-normal text-base md:text-xl"
              href="https://twitter.com/vicshi97"
              rel="noopener noreferrer"
              target="_blank"
            >
              Twitter
            </a>
            , and my
             
            <span
              class="underline-primary font-normal text-base md:text-xl"
            >
              portfolio website
            </span>
             
            (coming soon!).
          </p>
        </div>
      </div>
    </div>
  </div>
</body>
`;
